sql/core/src/test/scala/org/apache/spark/sql/MathFunctionsSuite.scala

```diff
@@ -262,6 +262,14 @@ class MathFunctionsSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("SPARK-44973 conv must allocate enough space for all digits plus negative sign") {
+    withSQLConf(SQLConf.ANSI_ENABLED.key -> false.toString) {
+      val df = Seq(("8" + "0"*15), ("-8" + "0" * 15)).toDF("num")
```

Comment:
It's interesting. I expect a linter error at `"0"*15` due to the lack of space, but it's passed.

In anyway, shall we change like this?
```
- "0"*15
+ "0" * 15
```

Reply:
I went with the `BigInt` proposal